snippets/python/cloud/cloud_assets_cleanup.py

```diff
@@ -0,0 +1,155 @@
+
+    if (method == "GET"):
+        response = requests.get(url, headers=headers, data = json.dumps(reqBody), verify=False)
+    return response
+
+# Login to specified NetBackup master server
+def loginMaster():
+    print ("** Logging in to the NetBackup master host...")
+    if "token" in globals:
+        return globals["token"]
+
+    reqBody = {
+     "userName" : args.nbu_user_name,
+        "password" : args.nbu_password
+    }
+    response = doRestCall("POST", buildURL(URL_LOGIN), reqBody)
+    if response.status_code == 201:
+        globals["token"] = response.json()["token"]
+        return response.json()["token"]
+    print("   -> Invalid user name or password")
+    print("** Exiting")
+    exit(1)
+
+# Get assets using a filter
+def getAssets():
+    token = loginMaster()
+    
+    print ("** Retrieving assets to cleanup...")
+    assets = []
+    cleanupTime = str((datetime.utcnow() - timedelta(hours = CLEANUP_TIME)).isoformat()) + "Z"
+    globals["cleanupTime"] = cleanupTime
+    count = 0
+    pageOffset = 0
+    while True:
+        queryFilter = "(lastDiscoveredTime  lt " + cleanupTime  \
+            + " and workloadType eq 'Cloud')"
+        queryFilter += "&page[offset]=" + str(pageOffset) + "&page[limit]=100"
+        response = doRestCall("GET", buildURL(URL_GET_ASSETS, queryFilter))
+        if response.status_code == 200:
+            if "data" not in response.json() or len(response.json()["data"]) == 0:
+                break
+            assets += response.json()["data"]
+            pageOffset += 100
+            if count == 0:
+                try:
+                    count = response.json()["meta"]["pagination"]["count"]
+                except:
+                    pass
+        else:
+            if len(assets) == 0:
+                print ("   -> No assets found")
+                print ("** Exiting")
+                exit(1)
+            else:
+                break
+        msg = str(len(assets))
+        if count != 0:
+            msg = str(len(assets)) + " / " + str(count)
+        print ("   -> Received " + msg + " stale assets")
+    globals["assets"] = assets
+
+def cleanupAssets():
+    assets = []
+    if "assets" in globals:
+        assets = globals["assets"]
+
+    if len(assets) == 0:
+        print("   -> No assets to clean up")
+        print("** Exiting")
+        exit(1)
+    
+    print ("** Cleaning up " + str(len(assets)) + " assets")
+
+    assetsCleanup = []
+    for asset in assets:
+        assetsCleanup.append(asset["id"])
+
+    reqBody = {"data": {
+        "type": "assetCleanup",
+        "id": "id",
+        "attributes": {
+            "cleanupTime": globals["cleanupTime"],
+            "assetIds": assetsCleanup
+            }
+        }
+    }
+    response = doRestCall("POST", buildURL(URL_ASSET_CLEANUP), reqBody)
+    if response.status_code == 204:
+        print ("** Assets cleaned up successfully")
+    else:
+        print("** Unable to clean assets")
```

Comment:
log received response code and message if possible.
Use logger except print statements

Reply:
I will incorporate this in next upcoming PR